### PR TITLE
fix nested syncing

### PR DESCRIPTION
**Pull request by koenvanderveen**

### CODE REVIEW

File: packages/syft/src/syft/client/client.py [hunk 1]
```diff
@@ -527,6 +528,29 @@ def create_project(
         project = project_create.start()
         return project
 
+    def sync_code_from_request(self, request):
+        code = deepcopy(request.code)
+        code.worker_pool_id = None
```
Comment (shubham3121) [inline packages/syft/src/syft/client/client.py]: `code.worker_pool_id` has been renamed to `code.worker_pool_name`.
Also, any reason we're setting worker_pool_id to None ? Don't we want the code to be executed on custom workers ?

File: packages/syft/src/syft/client/client.py [hunk 2]
```diff
@@ -527,6 +528,29 @@ def create_project(
...
+
+        def get_nested_codes(code):
+            result = []
+            for __, (linked_code_obj, _) in code.nested_codes.items():
+                nested_code = linked_code_obj.resolve
+                nested_code = deepcopy(nested_code)
+                nested_code.worker_pool_id = None
```
Comment (shubham3121) [inline packages/syft/src/syft/client/client.py]: Same here `worker_pool_id` attribute renamed to `worker_pool_name` in UserCode and SubmitUserCode

File: packages/syft/src/syft/client/client.py (hunk 1 above)
Comment (koenvanderveen) [reply packages/syft/src/syft/client/client.py]: no but there is a difference between the low side name/id and high side

File: packages/syft/src/syft/client/client.py [hunk 3]
```diff
@@ -527,6 +528,27 @@ def create_project(
         project = project_create.start()
         return project
 
+    def sync_code_from_request(self, request):
+        code = deepcopy(request.code)
```
Comment (shubham3121) [inline packages/syft/src/syft/client/client.py]: [Nitpick] Can we also check if request.code is a UserCode obj and not None or SyftError

**Review by shubham3121 — APPROVED** [review]
Lgtm !! 🎉 Just left a minor comment.

**Pull request merged**